pwndbg/gdblib/bpoint.py

```diff
@@ -0,0 +1,26 @@
+import gdb
+
+import pwndbg.lib.cache
+
+
+class Breakpoint(gdb.Breakpoint):
+    """
+    Breakpoint class, similar to gdb.Breakpoint, but clears the caches
+    associated with the stop event before determining whether it should stop the
+    inferior or not.
+
+    Unlike gdb.Breakpoint, users of this class should override `should_stop()`,
+    instead of `stop()`, as the latter is used to do cache invalidation.
+    """
+
+    def stop(self):
+        # Clear the cache for the stop event.
```

Comment:
Hmm we actually only clear cache for register values and not all caches that are normally cleared on stop event

Reply:
Fwiw this is where we set clearing caches for events: https://github.com/pwndbg/pwndbg/blob/e37591b25d923b1933b7092cacafa51a210d5f5e/pwndbg/gdblib/hooks.py#L51-L70

Reply:
I will just merge this and fixup this in another PR